.github/workflows/swan_pr_workflow.yml

```diff
@@ -0,0 +1,166 @@
+    types: [ created ] # Triggers the workflow when a release is created.
+
+# Defines the jobs that the workflow will execute
+jobs:
+  Lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install black
+        run: |
+          pip3 install black
+      - name: Lint with Black
+        run: |
+          black --check ./**/*.py
+
+  TypeCheck:
+    needs: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install mypy and type stubs
+        run: |
+          pip3 install mypy
+          pip3 install types-requests
+      - name: Type Check with Mypy
+        run: |
+          mypy --ignore-missing-imports ./**/*.py
+
+  Test:
+    needs: TypeCheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install pytest
+        run: |
+          pip3 install -r requirements.txt
+      - name: Run tests
+        run: pytest
+
+  Build:
+    needs: Test
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/dev'
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Cleanup /dist directory
+        run: |
+          rm -rf ./dist/*
+
+      - name: Build the project
+        run: |
+          python setup.py sdist bdist_wheel
+# UNCOMMENT ONLY AFTER PM DECISION
+  Deployments:
+    needs: Build
+    runs-on: ubuntu-latest
+    if: github.event_name == 'release' && github.event.action == 'created' && (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/dev')
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download dependencies artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: dependencies
+      - name: Load dependencies from cache
+        run: |
+          python -m pip3 install --no-index --find-links=dependencies twine
+      - name: Publish to PyPI
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+        run: |
+          twine upload dist/*
+
+# UNCOMMENT ONLY AFTER PM DECISION (Pluto)
+#  # Database setup job
+#  setup-database:
+#    runs-on: ubuntu-latest
+#    services: # Define the PostgreSQL service
+#      postgres:
+#        image: postgres:latest
+#        env:
+#          POSTGRES_DB: test_db  # Database name
+#          POSTGRES_USER: test_user  # Desired username
+#          POSTGRES_PASSWORD: test_password  # Desired password
+#        ports:
+#          - 5432:5432
+#    steps:
+#      - name: Wait for PostgreSQL to start
+#        run: |
+#          for i in {1..30}; do
+#            nc -z localhost 5432 && break
+#            sleep 1
+#          done
+#          sleep 10
+#
+#      - name: Install Python Dependencies
+#        run: |
+#          sudo apt-get -y install python3-pip3
+#          pip3 install --upgrade pip3
+#
+#      - name: Database Setup Steps
+#        run: |
+#          python setup_database.py
```

Comment:
I need to review this file before we UNCOMMENT this code

Reply:
its not so important you can set the followup ticket in several weeks to revise